web/packages/api/src/toEthereum_v2.ts

```diff
@@ -347,38 +374,84 @@ export async function getDeliveryFee(
         snowbridgeDeliveryFeeDOT = BigInt(leFee.toString())
     }
 
-    const xcm = buildResultXcmAssetHubERC20TransferFromParachain(
-        assetHub.registry,
-        registry.ethChainId,
-        "0x0000000000000000000000000000000000000000000000000000000000000000",
-        "0x0000000000000000000000000000000000000000",
-        "0x0000000000000000000000000000000000000000",
-        "0x0000000000000000000000000000000000000000000000000000000000000000",
-        340282366920938463463374607431768211455n,
-        340282366920938463463374607431768211455n,
-        340282366920938463463374607431768211455n,
-        parachain,
-        340282366920938463463374607431768211455n
-    )
+    const { tokenErcMetadata, sourceParachain, ahAssetMetadata, sourceAssetMetadata } =
+        resolveInputs(registry, tokenAddress, parachain)
 
-    let assetHubExecutionFeeDOT = 0n
-    let returnToSenderExecutionFeeDOT = 0n
-    let returnToSenderDeliveryFeeDOT = 0n
-    const bridgeHubDeliveryFeeDOT = await calculateDeliveryFee(
-        assetHub,
-        registry.bridgeHubParaId,
-        xcm
-    )
-    if (parachain !== registry.assetHubParaId) {
-        const returnToSenderXcm = buildParachainERC20ReceivedXcmOnDestination(
+    let xcm: any
+
+    if (sourceAssetMetadata.location) {
+        xcm = buildResultXcmAssetHubPNATransferFromParachain(
             assetHub.registry,
             registry.ethChainId,
+            sourceAssetMetadata,
```

Comment:
Since this is just an estimate of the fee, you do not need to pass in the asset token and rather just pass in a dummy token address or multilocaiton.

Reply:
I prefer to use the actual value if possible.